### PR TITLE
MM-59540 Ensure user has invite team permission in order to change setting

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
A Team Admin who has had the InviteUser Permission removed, could set the team setting 'AllowOpenInvites' thereby allowing anyone to join the team uninvited.

This PR checks for that permission if the `AllowOpenInvites` or `AllowDomains` field is being patched.

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-59540

#### Screenshots

Screenshot 2024-10-09 at 1 51 12 PM
Screenshot 2024-10-09 at 1 50 10 PM

#### Release Note

```release-note

```

### CODE REVIEW

File: webapp/channels/src/components/team_settings/team_access_tab/invite_section_input.tsx [hunk 1]
```diff
@@ -54,9 +51,6 @@ const InviteSectionInput = ({regenerateTeamInviteId}: Props) => {
         }
     }, [regenerateTeamInviteId, team?.id]);
 
-    if (!canInviteTeamMembers) {
-        return null;
-    }
```
Comment (sbishel) [inline webapp/channels/src/components/team_settings/team_access_tab/invite_section_input.tsx:59]: The entire tab is hidden, no reason to check this separately.

**Review by esarafianou — APPROVED** [review]

**Review by enahum — APPROVED** [review]

**Review by JulienTant — APPROVED** [review]

**Review by lindy65 — APPROVED** [review]
Thanks @sbishel - looks good to me 👍

**Comment by esarafianou** [comment]
@sbishel can we merge?

**Pull request merged**